detox/src/ios/system.js

```diff
@@ -0,0 +1,125 @@
+const assert = require('assert');
+
+const { DetoxRuntimeError } = require('../errors');
+const { systemActionDescription, expectDescription } = require('../utils/invocationTraceDescriptions');
+const log = require('../utils/logger').child({ cat: 'ws-client, ws' });
+const traceInvocationCall = require('../utils/traceInvocationCall').bind(null, log);
+
+
+class SystemExpect {
+  constructor(xcuitestRunner, element) {
+    this._xcuitestRunner = xcuitestRunner;
+    this.element = element;
+    this.modifiers = [];
+  }
+
+  toExist() {
+    const traceDescription = expectDescription.toExist();
+    return this.expect('toExist', traceDescription);
+  }
+
+  get not() {
+    this.modifiers.push('not');
+    return this;
+  }
+
+  createInvocation(systemExpectation) {
+    return {
+      type: 'systemExpectation',
+      systemPredicate: this.element.matcher.predicate,
+      ...(this.element.index !== undefined && { systemAtIndex: this.element.index }),
+      ...(this.modifiers.length !== 0 && { systemModifiers: this.modifiers }),
+      systemExpectation
+    };
+  }
+
+  expect(expectation, traceDescription) {
+    assert(traceDescription, `must provide trace description for expectation: \n ${JSON.stringify(expectation)}`);
+
+    const invocation = this.createInvocation(expectation);
+    traceDescription = expectDescription.full(traceDescription, this.modifiers.includes('not'));
+    return _executeInvocation(this._xcuitestRunner, invocation, traceDescription);
+  }
+}
+
+class SystemElement {
+  constructor(xcuitestRunner, matcher, index) {
+    this._xcuitestRunner = xcuitestRunner;
+    this.matcher = matcher;
+    this.index = index;
+  }
+
+  atIndex(index) {
+    if (typeof index !== 'number' || index < 0) throw new DetoxRuntimeError(`index should be an integer, got ${index} (${typeof index})`);
+    this.index = index;
+    return this;
+  }
+
+  tap() {
+    const traceDescription = systemActionDescription.tap();
+    return this.withAction('tap', traceDescription);
+  }
+
+  withAction(action, traceDescription) {
+    assert(traceDescription, `must provide trace description for action: \n ${JSON.stringify(action)}`);
+
+    const invocation = {
+      type: 'systemAction',
+      systemPredicate: this.matcher.predicate,
```

Comment:
Not something to fix, but rather out of curiosity? Was there a really good reason to prefix literally every property in the invocation object with `system` (systemThis, systemThat)? Didn't that complicate the implementation in some places?

Reply:
Okay, sorry, found your answer.